### PR TITLE
Version 2.0

### PR DESCRIPTION
**Pull request by Tranz5**
Changes for more stable block chain.
2min block time
1 day old min age
Old Clients will be disconnted on Friday Sep 15th

### CODE REVIEW

**Comment by Tranz5** [comment]
Let me know if there are any issues. Or please pull in as soon as you are able to.

Thanks!

File: src/main.cpp [hunk 1]
```diff
@@ -84,6 +82,23 @@ int64_t nSplitThreshold = GetProofOfWorkReward();
 int64_t nCombineThreshold = GetProofOfWorkReward() * 2;
 extern enum Checkpoints::CPMode CheckpointsMode;
 
+unsigned int SetTargetSpacing()
+{
+   unsigned int nStakeTargetSpacingSwitch = (fTestNet ? 1 * 60 : 1 * 30); // block spacing, 1 min tesnet, 30 seconds Main (OLD)
```
Comment (ahmedbodi) [inline src/main.cpp]: Wouldn't GetTargetSpacing be a better name?

File: src/main.cpp [hunk 2]
```diff
@@ -84,6 +82,23 @@ int64_t nSplitThreshold = GetProofOfWorkReward();
...
+   if ( nBestHeight + 1 > (fTestNet ? SPACING_TARGET_SWITCH_TESTNET : SPACING_TARGET_SWITCH))
+       nStakeTargetSpacingSwitch = 4.20 * 30; // 2 Minute 06 Second block spacing (NEW)
+
+   return nStakeTargetSpacingSwitch;
+}
+
+unsigned int SetStakeMinAge()
+{
+   unsigned int nStakeMinAgeSwitch = (fTestNet ? 2 * 60 * 60 : 60 * 60 * 24 * 10); // minimum age for coin age - 2hrs TestNet, 10 days Main (OLD)
+   if (nBestHeight + 1 > (fTestNet ? SPACING_TARGET_SWITCH_TESTNET : SPACING_TARGET_SWITCH))
```
Comment (ahmedbodi) [inline src/main.cpp]: Same as above

**Comment by ahmedbodi** [comment]
aside from the SetTarget functions which i think would be better called GetTarget it all looks good to me.

**Comment by Tranz5** [comment]
Thanks ahmed. I am going to update a few other things here in a few days I will put that on the list..

**Comment by Tranz5** [comment]
Alright I have updated to 2.0.1.0. Should take care of those renames and added a few other features and fixes.

**Comment by hclivess** [comment]
Are you guys still alive?

**Comment by ahmedbodi** [comment]
yuppp

**Comment by hclivess** [comment]
Hi Ahmed. Feel free to test my version and if you find it ok, we can merge it back here. A second opinion is always welcome. It has some additional features like instant auto snapshot syncing which I am not sure if are in alignment with the original vision with regards to security etc.

**Pull request merged**